internal/controller/certificate_controller.go

```diff
@@ -79,6 +83,12 @@ type CertificateReconciler struct {
 func (r *CertificateReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.Certificate{}).
+		Owns(&corev1.Secret{}).
+		WithEventFilter(predicate.Funcs{
+			DeleteFunc: func(e event.DeleteEvent) bool {
+				return true
+			},
```

Comment:
I could be wrong but I'm not sure this actually does any filtering. I _think_ that if you want to filter out `CREATE` and `UPDATE` events then you need to return a false on their predicate